sdk/core/Azure.Core/tests/samples/EventSamples.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using NUnit.Framework;
+
+namespace Azure.Core.Samples
+{
+    public class EventSamples
+    {
+        public class AlarmClientOptions : ClientOptions { }
+
+        public class AlarmClient
+        {
+            private ClientDiagnostics _clientDiagnostics = new ClientDiagnostics(new AlarmClientOptions());
+
+            public event SyncAsyncEventHandler<SyncAsyncEventArgs> Ring;
+
+            public void Snooze(CancellationToken cancellationToken = default) =>
+                SnoozeInternal(true, cancellationToken).GetAwaiter().GetResult();
+
+            public async Task SnoozeAsync(CancellationToken cancellationToken = default) =>
+                await SnoozeInternal(false, cancellationToken).ConfigureAwait(false);
+
+            protected virtual async Task SnoozeInternal(bool runSynchronously, CancellationToken cancellationToken)
+            {
+                // Why does snoozing an alarm always wait 9 minutes?
```

Comment:
Some theories say because, on average, in 10 minutes most people enter deeper sleep and wake up drowsy again; however, consensus is trending toward this: with more mechanical clocks, programming 2 digits (e.g. 10 minutes) is harder than a single digit, so engineers opted for efficiency.

Reply:
> _I'm not lazy; I'm efficient._
> 
> \- Abraham Lincoln, probably